Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -958,6 +968,20 @@ public boolean chooseTargetAmount(Outcome outcome, TargetAmount target, Ability
 
         // 2. Distribute amount between selected targets
 
+        // if only one target, it gets full amount, no possible choice
+        if (targets.size() == 1) {
+            target.setTargetAmount(targets.get(0), amountTotal, source, game);
+            return true;
+        }
+
+        // if number of targets equal to amount, each get 1, no possible choice
+        if (targets.size() == amountTotal) {
```

Comment:
if `targets.size() > amountTotal` then it's illegal and must cancel?

Reply:
It's true that is illegal. I think my code prevents that from occurring in the first place, but in case it somehow does, I can add an explicit check here (which would prevent it from getting stuck in the distribute dialog).